pyop2/base.py

```diff
@@ -3997,6 +3784,11 @@ def _jitmodule(self):
     def compute(self):
         """Executes the kernel over all members of the iteration space."""
         with timed_region("ParLoopExecute"):
+            lgmaps = []
+            for arg in self.args:
+                if arg._is_mat and arg.lgmaps is not None:
+                    lgmaps.append(arg.data.handle.getLGMap())
+                    arg.data.handle.setLGMap(*arg.lgmaps)
```

Comment:
The only problem with this is that PETSc barfs if the block size changed. So I think that means when we make a *matrix* we have to say what boundary conditions we will need (should it be block or unwound) and use the appropriate block size there.